src/project/ProjectModal.js

```diff
@@ -0,0 +1,143 @@
+                        uri: project.data.image && 
+                                    dashedHostname(project.data.image),
+                    }}
+                    style={style.image}
+                />
+
+                <Text children={project.data.name} 
+                    style={style.name} />
+                <Text 
+                    style={style.lastClaim} 
+                    children={'Your last claim submitted on 05-05-18'}
+                />
+            </View>
+            <ProjectActionButton
+                text='View Claim Forms'
+                icon={<MenuIcon fill='white'/>}
+            />
+            <ProjectActionButton
+                text='Disconnect from this Project'
+                icon={<LinkOffIcon fill='white'/>}
+                onPress={() => {
+                    Alert.alert('You sure?', '', [{
+                        text: 'Yes, delete',
+                        onPress: () => {
+                            ps.disconnect(projectId)
+                            onClose()
+                        },
+                    }, {
+                        text: 'Cancel',
+                        style: 'cancel',
+                    }])
+                }}
+            />
+            <ProjectActionButton
+                text='Turn off Project Notifications'
+                icon={<BellOffIcon fill='white'/>}
+            />
+            <ProjectActionButton
+                text='View the Project Page'
+                icon={<WebIcon fill='white'/>}
+                onPress={() => Linking.openURL(projectURL)}
+            /> 
+        </View>
+        <Button 
+            text='Close' 
+            type='contained'
+            size='lg' 
+            color='secondary' 
+            style={style.closeBtn}
+            onPress={onClose}
+        />
+    </Modal>
+}
+
+const style = StyleSheet.create({
+    root: {
+        flex: 1,
+        backgroundColor: 'rgba(0, 34, 51, 0.9)',
+        justifyContent: 'center',
+        padding: theme.spacing(3),
+    },
+    header: {
+        padding: theme.spacing(1.5),
+        marginBottom: theme.spacing(3),
+    },
+    image: {
+        width: '100%', 
+        height: 150, 
+        marginBottom: theme.spacing(2),
+    },
+    name: {
+        fontWeight: 'bold', 
+        color: 'white', 
+        fontSize: theme.fontSizes.h4,
+        marginBottom: theme.spacing(1),
+    },
+    lastClaim: {
+        fontSize: theme.fontSizes.p1, 
+        color: '#83D9F2',
+    },
+    closeBtn: {
+        borderRadius: 0, 
+        alignItems: 'center',
+    },
+})
+
+const ProjectActionButton = ({icon, text, onPress}) => {
+    return <Button 
+        text={text} 
+        size='lg'
+        color='secondary'
+        style={actionButtonStyles.root}
+        prefix={icon}
+        onPress={onPress}
+    />
+}
+
+const actionButtonStyles = StyleSheet.create({
+    root: {
+        marginBottom: theme.spacing(1),
+    },
+})
+
+const dashedHostname = urlStr =>
```

Comment:
I did not want to create a util file just for this function therefore I copy-pasted and use it here.

Reply:
I don't think it's a big deal at this point, but I would've preferred to reuse that function myself the time it has more than 1 caller. Up to you.